### ISSUE TITLE
Wildcards content awkwardly split between episodes 3 and 4

### ISSUE THREAD

**Issue by colinmorris** [swcarpentry/shell-novice#955]
Back when #699 was filed, we had an issue where an exercise in episode 3 required wildcards, but they were introduced and covered at length in episode 4.

The needle has since swung in the other direction, with the current split looking like:
- [Episode 3](http://swcarpentry.github.io/shell-novice/03-create/index.html): Introduces wildcards in a section at the end called 'Using wildcards for accessing multiple files at once', which comes with two wildcard exercises.
- [Episode 4](http://swcarpentry.github.io/shell-novice/04-pipefilter/index.html): Uses wildcard expressions throughout while demonstrating pipes and filters. Ends with a short digression on wildcards (starting with "Those numbers look good — but what’s that ‘Z’ doing there") and two more wildcard exercises.

I think we should move the wildcard-only content from episode 4 to 3, so it's all in one place. Episode 3 has wildcards listed in its 'key points' and episode 4 doesn't.

Some other suggestions re wildcard refactoring:
- I don't think the introduction to wildcards in episode 3 should be contained in a callout box (since it's part of the core lesson, rather than an aside).
- I'd personally be okay with just removing the episode 4 content on using the square bracket syntax for character sets. It just seems to me like a nice-but-niche trick that learners can live without. If we are going to keep it, I think it bears a bit more explanation (the way it's invoked in episode 4 currently almost makes it seem like it's a given, rather than a new concept).

**Comment by nayleeoid** [comment]
I agree the content on wildcards feels disjointed, particularly having them used fluently (but not discussed) all through lesson 4 and then suddenly having some exercises on them at the end.

Another solution would be to move all wildcard content to episode 4.  Wildcards to me are conceptually more akin to filters/pipes than they are to the basic file manipulations presented in episode 3.  That's why they feel tacked on in 3.  We could have wildcards be a key point in episode 4 and consolidate their content.  Introduce them at the top, continue to utilize their basic function, and have advanced content and exercises at bottom.  Another issue is that currently episode 3 is longer than 4, so shifting content from 3 to 4 will likely not produce a greater imbalance.

I also agree that the square brackets are being introduced in a "this should be obvious" sort of way, and that they could be left out altogether.

**Comment by gcapes** [comment]
Some rearrangement is probably in order (e.g. 'removing unneeded files' could go from ep4 to ep3), but the square bracket notation `*[AB]` is relied on in episode 5 and 6 to select the right files. That's not to say that those parts can't be rewritten as `NENE*A NENE*B` but just to point out the implications. I agree that the introduction of this square bracket notation if we keep it, should be improved.

Moving the wildcard callout content into the main body of text certainly sounds like a good idea.
